controller/common/manage_children.go

```diff
@@ -95,6 +98,26 @@ func revertObjectMetaSystemFields(newObj, orig *unstructured.Unstructured) error
 	return nil
 }
 
+// revertStatus overwrites the .status (i.e. all status fields) in newObj to match
+// what it was in orig by resetting .status directly without peeking into each field.
+func revertStatus(newObj, orig *unstructured.Unstructured) error {
```

Comment:
Optional suggestion: You could change `revertStatus` to `revertField` and add a `fieldPath ...string` parameter at the end. Then this becomes a generic function for reverting anything, and you can call it from inside the `revertObjectMetaSystemFields` function to avoid duplication.

Reply:
Yes, but this will remove your comments in `revertObjectMetaSystemFields` function. I will introduce the `revertField` function.

Reply:
Since `revertObjectMetaSystemFields` has more descriptive comments, you can copy those over to the new `revertField`. I believe they still apply.

Reply:
You are right, the comments still apply. Thanks!